### PR TITLE
DOC: More info about gRPC in v0.1

### PR DESCRIPTION
**Pull request by kikogolk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -470,7 +470,66 @@ message CountItemResponse {
 }
 ```
 
+As it was commented in the 'State of the Art' chapter, proto files are IDLs that define a interface. This proto files must be generated to real code using special tools. For example, as we are using Java in our Spring services, we decided to use **java-grpc** for this purpose.
```
Comment (alb3rtobr) [inline README.md]: "transformed" instead of "generated"?

File: README.md [hunk 2]
```diff
@@ -470,7 +470,66 @@ message CountItemResponse {
...
+
+In order to generate proto files to Java code it is needed to add the following dependencies in the `pom.xml` file of our maven project:
```
Comment (alb3rtobr) [inline README.md]: same comment. Or "In order to generate Java code from proto files"

**Review by alb3rtobr — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -470,7 +470,66 @@ message CountItemResponse {
...
+      <version>1.19.0</version>
+    </dependency>
+    <dependency>
+      <groupId>io.grpc</groupId>
+      <artifactId>grpc-stub</artifactId>
+      <version>1.19.0</version>
+    </dependency>
+```
+
+In addition it is needed to put our proto files inside `src/main/proto` directory and configure Maven to use the `Maven Protocol Buffers Plugin` to generate the java code from the proto files:
```
Comment (alb3rtobr) [inline README.md]: "Java"

File: README.md [hunk 4]
```diff
@@ -470,7 +470,66 @@ message CountItemResponse {
...
+              <goal>compile-custom</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+  </plugins>
+</build>
+```
+
+Once the java generated code is available, we may make use of the generated stubs to develop code in the clients and implement the provided interfaces for the services with the business logic in the server. The `dal` microservice will provide a server with the `ItemPersistenceService` service and `restapi` and `stockchecker` will implement a client to communicate with it.
```
Comment (alb3rtobr) [inline README.md]: "Java"

**Review by alb3rtobr — CHANGES_REQUESTED** [review]
There is also other "java" in the "3.3.2. H2 & MySQL" chapter

File: README.md (hunk 1 above)
Comment (kikogolk) [reply README.md]: they always says "generated code"

File: README.md (hunk 2 above)
Comment (kikogolk) [reply README.md]: ok!

File: README.md (hunk 3 above)
Comment (kikogolk) [reply README.md]: done

File: README.md (hunk 4 above)
Comment (kikogolk) [reply README.md]: done

**Review by alb3rtobr — APPROVED** [review]

**Pull request merged**